src/Routes/CreateServiceAccountPage/CreateModal.tsx

```diff
@@ -18,56 +18,93 @@ import { appendTo } from '../../shared/utils';
 
 const FORM_ID = 'service-account-form';
 const MAX_SERVICE_ACCOUNT_NAME_LENGTH = 32;
-const HELPER_TEXT = {
+const MAX_SERVICE_ACCOUNT_DESC_LENGTH = 255;
+const HELPER_TEXT: { [key: string]: string } = {
   'empty-name':
     'Must start with a letter and end with a letter or number. Valid characters include lowercase letters from a to z, numbers from 0 to 9, and hyphens ( - ).',
   'invalid-format':
     'Must start with a letter and end with a letter or number. Valid characters include lowercase letters from a to z, numbers from 0 to 9, and hyphens ( - ).',
   'invalid-length': `Cannot exceed ${MAX_SERVICE_ACCOUNT_NAME_LENGTH} characters.`,
+  'invalid-desc-length': `Cannot exceed ${MAX_SERVICE_ACCOUNT_DESC_LENGTH} characters.`,
 };
 
 export type CreateModalProps = {
   isCreating: boolean;
-  onSubmit: (name: string) => void;
+  onSubmit: (name: string, description: string) => void;
 };
 
 export const CreateModal: VoidFunctionComponent<CreateModalProps> = ({
   isCreating,
   onSubmit,
 }) => {
-  const [validate, setValidate] = useState(false);
+  type Validate = 'default' | 'error' | 'success';
   const [name, setName] = useState('');
+  const [description, setDescription] = useState('');
+  const [inputFieldBlur, setinputFieldBlur] = React.useState(false);
+  const [nameValidated, setNameValidated] = React.useState<Validate>('default');
+  const [descriptionValidated, setDescriptionValidated] =
+    React.useState<Validate>('default');
+  const [nameHelperText, setNameHelperText] = React.useState<string>('');
+  const [descHelperText, setDescHelperText] = React.useState<string>('');
+
   const submitButton = useRef();
 
-  const validity = React.useMemo(() => {
+  const isNameValid = (name: string) => {
     if (name === undefined || name.trim() === '') {
       return 'empty-name' as const;
     } else if (!/^[a-z]([-a-z0-9]*[a-z0-9])?$/.test(name.trim())) {
       return 'invalid-format' as const;
-    }
-    //validate max length
-    else if (name.length > MAX_SERVICE_ACCOUNT_NAME_LENGTH) {
+    } else if (name.length > MAX_SERVICE_ACCOUNT_NAME_LENGTH) {
       return 'invalid-length' as const;
+    } else {
+      return 'valid' as const;
+    }
+  };
+
+  React.useEffect(() => {
+    if (isNameValid(name) === 'valid') {
+      setNameValidated('success');
+      setNameHelperText(HELPER_TEXT['empty-name']);
+    } else if (name != '') {
+      setNameValidated('error');
+      setNameHelperText(HELPER_TEXT[isNameValid(name)]);
+    } else if (name == '' && inputFieldBlur) {
+      setNameValidated('error');
+      setNameHelperText(HELPER_TEXT[isNameValid(name)]);
+    }
+  }, [name, inputFieldBlur]);
```

Comment:
the useEffects could be simplified, something like...

```suggestion
  React.useEffect(() => {
    const validationResult = isNameValid(name);
    if (validationResult === 'valid') {
      setNameValidated('success');
      setNameHelperText(HELPER_TEXT['empty-name']);
    } else if (name != '' || name == '' && inputFieldBlur) {
      setNameValidated('error');
      setNameHelperText(HELPER_TEXT[validationResult]);
    }
  }, [name, inputFieldBlur]);
```